tasks/util/Publisher.ts

```diff
@@ -32,189 +27,116 @@ export function setConfig(key: string, value: string, global: boolean = false) {
-			exec('git rm -rf .', { silent: true, cwd: this.cloneDir });
+			exec(`git checkout --orphan ${ publishBranch }`, { silent: true, cwd: this.cloneDirectory });
+			exec('git rm -rf .', { silent: true, cwd: this.cloneDirectory });
 			this.log.writeln(`Created ${publishBranch} branch`);
 		}
 	}
 
 	/**
-	 * @return {boolean} indicates whether doc updates should be pushed to the origin
-	 */
-	shouldPush() {
-		const branch = process.env.TRAVIS_BRANCH || exec('git rev-parse --abbrev-ref HEAD', { silent: true }).trim();
-		return branch === 'master';
-	}
-
-	/**
-	 * If configuration information exists for obtaining a deployment key and prerequisites have been met to publish
-	 */
-	private canPublish(): boolean {
-		const skipDeploymentCredentials = typeof this.deployKey === 'boolean' && !this.deployKey;
-		return (skipDeploymentCredentials || this.hasDeployCredentials()) && this.shouldPush();
-	}
-
-	/**
-	 * Remove everything in preparation for the typedoc
-	 */
-	private refreshTypeDoc() {
-		const publishDir = this.subDirectory;
-
-		rm('-rf', join(this.cloneDir, publishDir));
-		cp('-r', this.generatedDocsDirectory, join(this.cloneDir, publishDir));
-	}
-
-	/**
-	 * Commit (but do not push) everything the new documentation
+	 * Publish the contents of { sourceDirectory } in the clone at { cloneDir } in the directory
+	 * { subDirectory } and push to the { branch } branch.
 	 */
-	private commit() {
-		const publishDir = this.subDirectory;
-
-		if (exec('git status --porcelain', { silent: true, cwd: this.cloneDir }) === '') {
-			this.log.writeln('Nothing changed');
-			return;
-		}
-
-		exec(`git add ${publishDir}`, { silent: true, cwd: this.cloneDir });
-		exec('git commit -m "Update API docs"', { silent: true, cwd: this.cloneDir });
+	publish() {
+		this.log.writeln(`Publishing ${this.branch} to origin`);
+		this.execSSHAgent('git', [ 'push', 'origin', this.branch ], { silent: true, cwd: this.cloneDirectory });
+		this.log.writeln(`Pushed ${this.branch} to origin`);
 	}
 
 	/**
 	 * Execute a credentialed git command
 	 * @param command the command to execute
 	 * @param options execute options
 	 */
-	private execSSHAgent(command: string, options: any = {}): string {
+	private execSSHAgent(command: string, args: string[] = [], options: any = {}): string {
 		if (this.hasDeployCredentials()) {
 			const deployKey: string = <string> this.deployKey;
 			const relativeDeployKey = options.cwd ? relative(options.cwd, deployKey) : deployKey;
 			chmodSync(deployKey, '600');
-			return exec(`ssh-agent bash -c 'ssh-add ${ relativeDeployKey }; ${ command }'`, options);
+			return exec(`ssh-agent bash -c 'ssh-add ${ relativeDeployKey }; ${ command } ${ args.join(' ') }'`, options);
 		}
 		else {
-			return exec(command, options);
+			this.log.writeln(`Deploy Key "${ this.deployKey }" is not present. Using environment credentials.`);
+			return spawn(command, args, options).stdout;
```

Comment:
This will execute commands in the current process/shell when a deploy key is not explicitly available